### ISSUE TITLE
go to definition does not work for implicit index.js files

### ISSUE THREAD

**Issue by a-x-** [microsoft/TypeScript#31632]
- [x] I checked similar issues.
- [x] I checked it with VS Code Insiders w/o extensions
- [x] I checked it w/o extensions

- VSCode Version: 1.34.0
- OS Version: macOS 10.14.5

Steps to Reproduce:

1. cmd-click onto js import

Yay, Look at Minimal repro repo: https://github.com/a-x-/chore-vscode-gotodef-bug
Start with [src/index.js](https://github.com/a-x-/chore-vscode-gotodef-bug/blob/master/src/index.js)
Then check jsconfig.json and src/b dir

**Comment by mjbvz** [comment]
Possibly #16577 

The root cause is that there is an import with an file extension:

```ts
export { default } from 'b.js'
```

Extensionless imports work as expected:

```ts
export { default } from 'b'
```

**Comment by RyanCavanaugh** [comment]
@mjbvz what is the bug here?

**Comment by mjbvz** [comment]
The example github repo seems fixes using TS 3.5.1 on the latest VS Code insiders. The only import that still doesn't resolve is:

```js
import {} from './b';
```

But i'm not sure if that's is by-design.

If you set `moduleResolution: node` as I first suggested, then `export { default } from 'b.js'` does not get resolved. Again I'm not sure if that is by-design or not
